content/blog/2025/01/08/2025-01-08-jenkins-2024-recap.adoc

```diff
@@ -0,0 +1,176 @@
+Their efforts may be unseen at times, but they are always felt throughout the project.
+
+[[security]]
+image:/images/post-images/2025/01/08/security.png[Security Update]
+
+At the beginning of 2024, link:https://www.jenkins.io/blog/2024/01/25/sonar-vulnerability-report/[Sonar worked with the Jenkins security team] to address a previously unnoticed vulnerability.
+The issue was very niche and would only execute given a specific set of circumstances and permissions, so there was no larger threat here.
+However, the fact that this vulnerability existed was enough reason for Jenkins security and Sonar to collaborate and resolve this issue.
+Thanks to their in-depth work and tenacity in removing the vulnerability, Jenkins continues to be a secure and well-maintained DevOps platform.
+
+Additionally, thanks to Alpha Omega for their link:https://www.jenkins.io/blog/2024/10/04/content-security-policy-grant/[grant and collaboration], we have been able to make a concerted effort to update the CSP for Jenkins.
+Their grant has allowed us to bring in two key contributors to take on the work of this project and their work has been nothing short of incredible.
+From the very start of the project, Shlomo Dahan and Yaroslav Afenkin, under the guidance of Basil Crow, have been working furiously to help improve the CSP of Jenkins and bring some of the most important plugins up to speed.
+
+By the numbers:
+
+* Published 6 security advisories for Jenkins core
+* Published 8 security advisories for 33 Jenkins plugins
+
+In addition to the collaboration with Sonar, the Jenkins Security team has worked constantly throughout 2024 to ensure that security issues are acknowledged and resolved whether they are for Jenkins core or plugins within the ecosystem.
+
+[[platform]]
+image:/images/post-images/2025/01/08/platform-modernization.png[Platform Modernization Update]
+
+Over the summer, the Jenkins weekly release line made the move to link:https://www.jenkins.io/blog/2024/06/11/require-java-17/[require Java 17] as a baseline.
+This was an important update to keep up with current usage and start the process of dropping support for Java 11.
+Thanks to the hard work and dedication of Jenkins developers, the transition was successful and we saw a steady rise in JDK 17 installation and usage.
+By the end of October LTS release, the Java 17 requirement had been thoroughly tested and used over the preceding months, ensuring that its inclusion would be safe.
+Since the October release of LTS 2.479.1, we have not seen any major issues with Java 17 usage.
+While there have been some hiccups when it comes to upgrading plugins alongside the Jenkins LTS, these have been addressed as they come up and typically can be resolved by ensuring the plugin is updated before and after upgrading Jenkins.
```

Comment:
```suggestion
By the end of the October LTS release, the Java 17 requirement had been thoroughly tested and used over the preceding months, ensuring that its inclusion would be safe.
Since the October release of LTS 2.479.1, we have not seen any major issues with Java 17 usage.
While there have been some hiccups when it comes to upgrading plugins alongside the Jenkins LTS, these have been addressed as they come up and typically can be resolved by ensuring the plugin is updated before and after upgrading Jenkins.
```